pkg/apis/authorization/types.go

```diff
@@ -177,6 +177,37 @@ type SelfSubjectRulesReviewSpec struct {
 	Namespace string
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SubjectRulesReview is like SelfSubjectRulesReview, but can query any user, not just the current user.
+type SubjectRulesReview struct {
```

Comment:
Note that this type wasn't implement as an API server resource and is currently only used for the wehbook. Would be open to adding this as a full fledged API resource.

Reply:
If we do that, it should be a little more locked down than the SelfSubjectAccessReview. The permissions around it should be more like get/list ClusterRoleBindings, since you're potentially seeing policy from levels above the resource you asked about.

I think it could be a valuable API for auditing for admin level users. E.g. `kubectl what-can-they-do --service-account default:default`. Worth a discussion outside of this PR.

Reply:
> I think it could be a valuable API for auditing for admin level users. E.g. kubectl what-can-they-do --service-account default:default. Worth a discussion outside of this PR.

We've got an API for that too if you'd like to port it: https://github.com/openshift/origin/blob/master/pkg/authorization/apis/authorization/types.go#L190-L198 :)

Reply:
> If we do that, it should be a little more locked down than the SelfSubjectAccessReview. The permissions around it should be more like get/list ClusterRoleBindings, since you're potentially seeing policy from levels above the resource you asked about.

Maybe locked down like SubjectAccessReview which allows similar powers?